mobile/test/favorite_provider_test.dart

```diff
@@ -0,0 +1,138 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/modules/favorite/providers/favorite_provider.dart';
+import 'package:immich_mobile/shared/models/asset.dart';
+import 'package:immich_mobile/shared/providers/asset.provider.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+
+@GenerateNiceMocks([
+  MockSpec<AssetsState>(),
+  MockSpec<AssetNotifier>(),
+])
+import 'favorite_provider_test.mocks.dart';
+
+Asset _getTestAsset(String id, bool favorite) {
+  return Asset(
+    remoteId: id,
+    deviceAssetId: '',
+    deviceId: '',
+    ownerId: '',
+    fileCreatedAt: DateTime.now(),
+    fileModifiedAt: DateTime.now(),
+    durationInSeconds: 0,
+    fileName: '',
+    isFavorite: favorite,
+  );
+}
+
+void main() {
+  group("Test favoriteProvider", () {
+    test("Empty favorites provider", () {
+      final container = ProviderContainer();
+
+      final assetsState = MockAssetsState();
```

Comment:
Can you put this in a test framework set up function that runs before each test?

Reply:
Then I'd have to make them global and I'd like to avoid that to avoid problems since different tests mock methods differently. I agree that some code is duplicated but keeping the tests independent from each other was my main priority here.

Reply:
Ok. That's how they are in the server codebase if you wanted to take a look. We instantiated the service and mocks automatically before each test but then set the different mocked responses in each test before invoking the service method.

Reply:
Thanks for the suggestion! I had a look at the server test and understand what you mean. Added it and it seems to work fine & reduces dup code.